pkg/controller/etcdproxy/helpers.go

```diff
@@ -0,0 +1,185 @@
+								{
+									Name:      etcdCoreCertSecretName,
+									MountPath: "/etc/certs/client",
+									ReadOnly:  true,
+								},
+								{
+									Name:      etcdCoreCAConfigMapName,
+									MountPath: "/etc/certs/ca",
+									ReadOnly:  true,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: etcdCoreCertSecretName,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: etcdCoreCertSecretName,
+								},
+							},
+						},
+						{
+							Name: etcdCoreCAConfigMapName,
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: etcdCoreCAConfigMapName,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newService(etcdstorage *samplev1alpha1.EtcdStorage,
+	etcdControllerNamespace string) *corev1.Service {
+	labels := map[string]string{
+		"controller": "epc",
+	}
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName(etcdstorage),
+			Namespace: etcdControllerNamespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(etcdstorage, schema.GroupVersionKind{
+					Group:   samplev1alpha1.SchemeGroupVersion.Group,
+					Version: samplev1alpha1.SchemeGroupVersion.Version,
+					Kind:    "EtcdStorage",
+				}),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol:   corev1.ProtocolTCP,
+					Port:       2379,
+					TargetPort: intstr.FromInt(2379),
+				},
+			},
+		},
+	}
+}
+
+// replicaSetName calculates name to be used to create a ReplicaSet.
+func replicaSetName(etcdstorage *samplev1alpha1.EtcdStorage) string {
+	return fmt.Sprintf("etcd-rs-%s", etcdstorage.ObjectMeta.Name)
+}
+
+// serviceName calculates name to be used to create a ReplicaSet.
+func serviceName(etcdstorage *samplev1alpha1.EtcdStorage) string {
+	return fmt.Sprintf("etcd-%s", etcdstorage.ObjectMeta.Name)
+}
+
+// flagfromString returns double dash prefixed flag calculated from provided key and value.
+func flagfromString(key, value string) string {
+	return fmt.Sprintf("--%s=%s", key, value)
+}
+
+// ControllerNamespace returns name of the namespace where controller is located. The namespace name is obtained
+// from the "/var/run/secrets/kubernetes.io/serviceaccount/namespace" file. In case it's not possible to obtain it
+// from that file, the function resorts to the default name, `kube-apiserver-storage`.
+func ControllerNamespace() string {
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
+			return ns
+		}
+	}
+
+	runtime.HandleError(fmt.Errorf("unable to detect controller namespace, resorting to default namespace name: kube-apiserver-storage"))
```

Comment:
I believe this is a bad error handling here. Should I use `glog` to log here, or some other function?